schemas/cloudflare/dns-record-1.yml

```diff
@@ -0,0 +1,33 @@
+---
+"$schema": /metaschema-1.json
+version: '1.0'
+type: object
+
+additionalProperties: false
+properties:
+  "$schema":
+    type: string
+    enum:
+    - /cloudflare/dns-record-1.yml
+  name:
+    type: string
+  'type':
+    type: string
+    enum:
+    - A
+    - CNAME
+    - TXT
+    - NS
+    - DS # The record used to identify the DNSSEC signing key of a delegated zone
```

Comment:
If we're supporting [DS and DNSKEY](https://github.com/cloudflare/terraform-provider-cloudflare/blob/02bf98d3154dec4fdaa5788d0a14c92733214ad6/internal/provider/schema_cloudflare_record.go#L176), will we need to also have a separate `data` field in the schema to map to the [Terraform provider schema field](https://registry.terraform.io/providers/cloudflare/cloudflare/latest/docs/resources/record#data)?

Reply:
Good catch!

Reply:
Add new CloudflareDnsRecordDataSettings_v1 schema to encapsulate all the data fields for DS and DNSKEY.